### PR TITLE
blakerc_codereview

### PR DESCRIPTION
**Pull request by Blakerc**
Analysis of the program. : This program seems to be made to organize your slice files for 3d printing. This is something that I didn't realize was a problem, but now that I think about it, makes a lot of sense.

Was the program available in UC Github on time? Yes

Is the program documented/commented well enough for you to understand? Yes the program makes a lot of sense and has a VERY well written back-end so far, the DTO's and DAO,s are structured in very understandable and logical ways

Does the program compile? Yes

Rationale behind your changes. There was very little changes that needed to be done, the only thing I could find was that they were using android.text.format.Time, which is actually depreciated as of API 22 and should be replaced with calendar or GregorianCalander, this will also allow for better access to functions revolving around the time column

Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.

https://github.com/Blakerc/MobileMeetings/commit/fd0919c1d0f03b15d1b5ed48fbed128d2e9a012a
https://github.com/Blakerc/MobileMeetings/commit/6424eafeb06b1b48f1a82d179ebbecaecad4a73a
https://github.com/Blakerc/MobileMeetings/commit/28381000433ebd4a4053f987e3a3e9d901678957

 
Next, list three specific technical concepts that you learned from this code review.  Copy and paste samples to demonstrate your point.  Examples:

  1. in their SliceFile.kt they show a great example of setting up a table with a PK and all of the columns, this is something i haven't done yet, but from what i can tell this is a very good example

@Entity(tableName = "SliceFile")
data class SliceFile (@PrimaryKey var sfid: Int,
                      @ColumnInfo(name = "sliceFileName") var sliceFileName : String,
                      @ColumnInfo(name = "modelId") var modelId : Int,

================

2. in the MainFragment they get the data ready to be loaded into autocompletes, this seems to be a streamlined way to do it, and different then how I've done it in the past

        applicationViewModel.printerService.getPrinterDAO().getAll().observe(viewLifecycleOwner, Observer {
//                printers ->
        })

===================

3. even though it's commented out, it's cool to see this test, which uses @before and @after to test if the SQL db has started ok

//    private lateinit var db: SliceDatabase
//    @Before
//    fun createDb() {
//        val context = ApplicationProvider.getApplicationContext<Context>()
//        db = Room.databaseBuilder(
//            context,
//            SliceDatabase::class.java, "main-db"
//        ).build()
//    }
//
////    @After
////    @Throws(IOException::class)
////    fun closeDb() {
////        db.close()
////    }

### CODE REVIEW

File: app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package edu.uc.kovaciad.slicetracker.dto
 
 import android.text.format.Time
+//as of api 22 Time is depreciated and it is suggested to use GregorianCalendar
```
Comment (Blakerc) [inline app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt:4]: I'm not sure if calendar or GregorianCalander is actually the best option, as you are using this variable to store an amount of time something will take, so there will probably have to be some conversions done

File: .idea/codeStyles/codeStyleConfig.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (Blakerc) [inline .idea/codeStyles/codeStyleConfig.xml:1]: This is a nice little intellij settingthat helps out with multiple developers working on the same code

File: app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt:4]: Putting conversation/discussion  in code comments is bad practice, as that clutters up the source code and does not belong in source code files.  I mentioned that in the code review instructions.    Discussion and comments are very good, but have that in the appropriate place, which is in the "Files Changed" tab in the pull request in GitHub (which is where I am writing this and you are reading this).  That way, you can have a conversation with replies, right where the code is happening, and without cluttering up the source code itself.

File: app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt [hunk 3]
```diff
@@ -37,6 +39,7 @@ data class SliceFile (@PrimaryKey var sfid: Int,
                       @ColumnInfo(name = "resinLiftHeight") var resinLiftHeight : Int = 0,
                       @ColumnInfo(name = "numberOfLayers") var numberOfLayers : Int = 0,
                       @ColumnInfo(name = "filamentNozzleThickness") var filamentNozzleThickness : Double = 0.0,
+                      //Time should be changed with GregorianCalander (I dont want to mess up your DB so i wont make this change)
```
Comment (discospiff) [inline app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt:42]: Putting conversation/discussion  in code comments is bad practice, as that clutters up the source code and does not belong in source code files.  I mentioned that in the code review instructions.    Discussion and comments are very good, but have that in the appropriate place, which is in the "Files Changed" tab in the pull request in GitHub (which is where I am writing this and you are reading this).  That way, you can have a conversation with replies, right where the code is happening, and without cluttering up the source code itself.

Comment (discospiff) [inline app/src/main/java/edu/uc/kovaciad/slicetracker/dto/SliceFile.kt:42]: On: " //Time should be changed with GregorianCalander (I dont want to mess up your DB so i wont make this change)"

You're in a separate branch.  You can make any change you want, without messing up the database.  That's why we use branches.  :)

**Comment by discospiff** [comment]
The changes in this branch are discussion, not source code changes, so I don't recommend merging this branch.

**Comment by kovaciad** [comment]
At the professor's suggestion, we will not be using this pull request.

**Pull request closed without merging**